webapp/channels/src/packages/mattermost-redux/src/reducers/entities/scheduled_posts.ts

```diff
@@ -175,8 +175,8 @@ function errorsByTeamId(state: ScheduledPostsState['errorsByTeamId'] = {}, actio
 function byChannelOrThreadId(state: ScheduledPostsState['byChannelOrThreadId'] = {}, action: MMReduxAction) {
     switch (action.type) {
     case ScheduledPostTypes.SCHEDULED_POSTS_RECEIVED: {
-        const {scheduledPostsByTeamId} = action.data;
-        const newState = {...state};
+        const {scheduledPostsByTeamId, prune} = action.data;
+        const newState = prune ? {} : {...state};
```

Comment:
when the reconnect happens, we specifically tell the store to prune the information about the scheduled messages by channel or thread id. This approach is simpler and safer than the previous I was trying.

Reply:
If the scheduled posts screen is mounted, this may cause all the selectors there to retrigger, since every scheduled post is now a new scheduled post.

Probably not a big issue (we won't have so many scheduled posts for this to be a problem, and the scheduled post render shouldn't be so heavy). But something I wanted to flag.

Reply:
ok, understood, there had to be a sacrifice somewhere. The good thing is that this is just for the byChannelOrThreadId scheduled posts, so, only that part will be affected. byId and byTeamId will remain the same. 👍